### PR TITLE
Bug 1482474 - Unicode characters in comments are garbled

### PR DESCRIPTION
**Pull request by dylanwh**

### CODE REVIEW

**Comment by kyoshino** [comment]
Hmm, I see the following internal error:

> undef error - Undefined subroutine &Bugzilla::Template::decode called

**Comment by dylanwh** [comment]
This depends on #711

File: Bugzilla/Template.pm [hunk 1]
```diff
@@ -270,12 +272,14 @@ sub renderComment {
...
-    }
-    else {
-        return Bugzilla->markdown_parser->render_html($text);
+    unless ($skip_markdown) {
+        $text = decode('UTF-8', Bugzilla->markdown_parser->render_html($text));
     }
+
+    my $dom = Mojo::DOM->new($text);
+    my $urlbase = Bugzilla->localconfig->{urlbase};
+    $dom->find("a[href]:not([href^=$urlbase])")->map(attr => rel => 'nofollow');
```
Comment (kyoshino) [inline Bugzilla/Template.pm]: Well, Unicode characters and external links are fine, but looks like I've found an issue: internal bug links like `Bug 123` also have `rel="nofollow"`.

Comment (kyoshino) [reply Bugzilla/Template.pm]: Also, not related to this PR, but I see a duplicated, empty `<a>` tag for the first bug link in a rendered comment. For example,

```
http://bmo-web.vm/show_bug.cgi?id=1 or Bug 1, Bug 2
```

turns into:

```html
<a class="bz_bug_link 
bz_status_NEW " title="NEW - First Love" href="show_bug.cgi?id=1"></a><a href="http://bmo-web.vm/show_bug.cgi?id=1">http://bmo-web.vm/show_bug.cgi?id=1</a> or <a class="bz_bug_link 
bz_status_NEW " title="NEW - First Love" href="show_bug.cgi?id=1">Bug 1</a>, <a class="bz_bug_link 
bz_status_NEW " title="NEW - Yo" href="show_bug.cgi?id=2">Bug 2</a>
```

🤔

Comment (dylanwh) [reply Bugzilla/Template.pm]: this has been tricky but I finally have a fix. will post in the morning

Comment (globau) [reply Bugzilla/Template.pm]: what sort of performance hit does running each comment through a DOM parser have?

**Pull request closed without merging**